### ISSUE TITLE
Add "AND/OR" switch to status filter 

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#11288]
### Tell us your idea!

Today, it was the first time I have applied multiple parameters in the status filter. (In my case "offline log" and "found") 

![Screenshot_20210721_200915_cgeo geocaching](https://user-images.githubusercontent.com/64581222/126548928-978421a4-7da8-4340-8574-832bda273614.jpg)

First, I was curious what will happen, but the result was pretty clear: both filters where joined with AND which resulted in 0 caches in my case.

Than I wanted to change this to OR. I tried out, if the "global OR setting" will affect this setting, but it isn't the case (which seems totally logical to me). 

### What solution would you suggest?

I would like to have a switch where I can choose if the parameters are connected via AND or OR. 

### Is there an alternative?

- Add the filter multiple times (a bit ugly as the filter is very large) 
- Split the status filter into many small filters (it's the only filter, where you can select many totally different things anyway)
- define both parameters with the opposite value and negate the whole filter (Requires a little thinking power ;)) 

### Additional Information

Somehow, but not really related to  #11276

**Comment by eddiemuc** [comment]
I agree that particularly the status filter grew bigger than maybe is good form him, and I also agree that this makes the solution to use multiple of it look ugly when using OR (which was the original idea behind the possibility to include same filter type multiple times).

I would vote for splitting up the filter, although I have no good idea into which groups and how to name them...

For the use case at hand: the status filter has an internal "basic" and "advanced" view, which could be made explicitly by providing a switch for this.

(An OR switch at this place would be hard to understand imho. For example would such a switch also "OR" the states active, disabled, archived?)

**Comment by fm-sys** [comment]
> An OR switch at this place would be hard to understand imho. For example would such a switch also "OR" the states active, disabled, archived?

I would have added a separator line and placed the switch below below the active, disabled, archived.

![Screenshot_20210722_083542](https://user-images.githubusercontent.com/64581222/126598724-969a01d4-24e8-478a-a33d-c60fe7f84144.png)

But you are right, maybe it's cleaner to split it up instead of making it even more complex...

What about a new filter filter category "status" with filters like "findability" (active, disabled, archived), "found state" (found and maybe logged offline + own + a new setting DNF), and so on... 

Then we could even avoid the internal "basic" and "advanced" view, as "basic" would get  "findability" and "found state" but not the other ones.

**Comment by MagpieFourtyTwo** [comment]
> What about a new filter filter category "status" with filters like "findability" (active, disabled, archived)

Looks pretty familiar ... ;)

**Comment by fm-sys** [comment]
No one's asked for this in years — including me — so as part of a cleanup round of old issues I created myself, I'm closing it now. One of those _just-in-case_ ideas that never went anywhere...
